File: ToolKit/src/main/java/com/uday/android/toolkit/ui/DialogUtils.java
package com.uday.android.toolkit.ui;
import android.content.*;
import android.app.*;
import android.widget.*;
import com.uday.android.toolkit.R;
import android.view.*;
import android.view.View.*;
import android.graphics.*;
import android.util.*;
import android.text.*;
import com.uday.android.toolkit.*;

public class DialogUtils
{
	private Context context;
	private AlertDialog mDialog;
	private AlertDialog.Builder dialogBuilder;
	private Button mPositive,mNegative,mNeutral;
	private View mView;
	
	public DialogUtils(Context context){
		this.context=context;
		dialogBuilder=new AlertDialog.Builder(context)
					.setPositiveButton("ok",null)
					.setTitle(" ").setMessage(" ")
					.setNegativeButton("cancel",null)
					.setNeutralButton("another",null);
	}
	
	public DialogUtils create(){
		mDialog=dialogBuilder.create();
		mDialog.getWindow().getAttributes().windowAnimations = R.style.DialogTheme;
		return this;
	}
	
	public DialogUtils setView(View view){
		mView=view;
		dialogBuilder.setView(view);
		return this;
	}
	
	public DialogUtils setView(int layoutRes){
		mView=((Activity)context).getLayoutInflater().inflate(layoutRes,null);
		dialogBuilder.setView(mView);
		return this;
	}
	
	public Button getPositiveButton(){
		return mPositive;
	}
	
	public Button getNegativeButton(){
		return mNegative;
	}
	
	public Button getNeutralButton(){
		if(mNeutral!=null){
			mNeutral.setVisibility(View.VISIBLE);
		}
		return mNeutral;
	}
	public AlertDialog getDialog(){
		return mDialog;
	}
	
	public DialogUtils setTitle(String title){
		mDialog.setTitle(title);
		return this;
	}
	
	public DialogUtils setMessage(String message){
		mDialog.setMessage(message);
		return this;
	}
	
	public DialogUtils show(){
		mDialog.show();
		if(mPositive ==null)
			mPositive=mDialog.getButton(DialogInterface.BUTTON_POSITIVE);
		if(mNegative==null)
			mNegative=mDialog.getButton(DialogInterface.BUTTON_NEGATIVE);
		if(mNeutral==null)
			mNeutral=mDialog.getButton(DialogInterface.BUTTON_NEUTRAL);
		return this;
	}
	
	public DialogUtils cancel(){
		mDialog.cancel();
		return this;
	}
	
	public DialogUtils showConfirmDialog(boolean retainView,int icRes,String title,String message,String cancelText,String confirmText,final OnClickListener confirmListener){
		if(mDialog==null)create();
		show();
		mNeutral.setVisibility(View.GONE);
		if(mView!=null && !retainView)mView.setVisibility(View.GONE);
		if(title!=null)mDialog.setTitle(title);else mDialog.setTitle("");
		if(message!=null)mDialog.setMessage(message);else mDialog.setMessage("");
		if(cancelText!=null)mNegative.setText(cancelText);else mNegative.setVisibility(View.GONE);
		if(confirmText!=null)mPositive.setText(confirmText);else mPositive.setVisibility(View.GONE);
		if(icRes!=0)mDialog.setIcon(icRes);else mDialog.setIcon(null);
		if(confirmListener!=null)mPositive.setOnClickListener(new View.OnClickListener(){
			@Override public void onClick(View p1){
				confirmListener.onClick(mDialog);
			}
		});
		
		return this;
	}
	
	public static EditText showEditTextDialog(Context context,String title,String message,String editorContent,String description,String confirmText,final DialogUtils.OnClickListener confirmListener){
		LinearLayout contentView=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.edit_text,null);
		EditText edtTxt=(EditText)contentView.findViewById(R.id.edt_txt);
		edtTxt.setText(editorContent);
		edtTxt.setHint(description);
		edtTxt.setSelectAllOnFocus(true);
		edtTxt.setSingleLine(true);
		final AlertDialog dialog=new AlertDialog.Builder(context)
							.setTitle(title)
							.setMessage(message)
							.setView(contentView)
							.setPositiveButton(confirmText,null)
							.setNegativeButton("cancel",null)
							.create();
			dialog.getWindow().getAttributes().windowAnimations = R.style.DialogTheme;
		dialog.show();
		dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener(){
			@Override
			public void onClick(View p1){
				confirmListener.onClick(dialog);
			}
		});
		
		return edtTxt;
	}
	
	public static Object[] showTermDialog(Context context,String title,String subtitle,String positive,String negative){
		LinearLayout titleView=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.term_dialog_header,null);
		((TextView)titleView.findViewById(R.id.term_title)).setText(title);
		((TextView)titleView.findViewById(R.id.term_dscrptn)).setText(subtitle);
		
		LinearLayout contentView=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.term_dialog_content,null);
		contentView.findViewById(R.id.term_scroller).setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,(int)(MainActivity.SCREEN_HEIGHT*0.5)));
		AlertDialog dialog=new AlertDialog.Builder(context)
				.setCustomTitle(titleView)
				.setView(contentView)
				.setNegativeButton(negative,null)
				.setPositiveButton(positive,null)
				.create();
		dialog.getWindow().getAttributes().windowAnimations = R.style.DialogTheme;
		dialog.show();
		dialog.setCancelable(false);
		dialog.getButton(DialogInterface.BUTTON_POSITIVE).setVisibility(View.GONE);
		dialog.getButton(DialogInterface.BUTTON_NEGATIVE).setEnabled(false);
		
		return new Object[]{dialog,contentView.findViewById(R.id.term_text),contentView.findViewById(R.id.term_progress)};
	}
	
	
	public static AlertDialog showConfirmDialog(Context context,String title,String content,Spanned spannedContent,String confirmTxt,final OnClickListener listener){
		AlertDialog.Builder dialogBuilder= new AlertDialog.Builder(context).setTitle(title)
							.setMessage(content)
							.setNegativeButton("cancel",null)
							.setPositiveButton(confirmTxt,null);
		if(spannedContent!=null){
			TextView tw=new TextView(context);
			tw.setPadding(40,15,20,0);
			tw.setText(spannedContent);
			dialogBuilder.setView(tw);
		}
		final AlertDialog dialog=dialogBuilder.create();		
		dialog.getWindow().getAttributes().windowAnimations = R.style.DialogTheme;
		dialog.show();
		if(listener!=null){
			dialog.getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener(){
				@Override
				public void onClick(View p1){
					listener.onClick(dialog);
				}
			});
		}
		return dialog;
	}
	
	public static interface OnClickListener{
		public void onClick(AlertDialog p1);
	}
}

File: app/src/main/java/com/uday/android/toolkit/ui/DialogUtils.kt
package com.uday.android.toolkit.ui

import android.annotation.SuppressLint
import android.content.Context
import android.content.DialogInterface
import android.text.Spanned
import android.view.View
import android.widget.*
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.uday.android.toolkit.MainActivity
import com.uday.android.toolkit.R

class DialogUtils(private val context: Context) {
    var dialog: AlertDialog? = null
        private set
    private val dialogBuilder: AlertDialog.Builder = AlertDialog.Builder(context)
        .setPositiveButton("ok", null)
        .setTitle(" ").setMessage(" ")
        .setNegativeButton("cancel", null)
        .setNeutralButton("another", null)
    var positiveButton: Button? = null
        private set
    var negativeButton: Button? = null
        private set
    private var mNeutral: Button? = null
    private var mView: View? = null

    val neutralButton: Button?
        get() {
            if (mNeutral != null) {
                mNeutral!!.visibility = View.VISIBLE
            }
            return mNeutral
        }

    fun create(): DialogUtils {
        dialog = dialogBuilder.create()
        dialog!!.window!!.attributes.windowAnimations = R.style.DialogTheme
        return this
    }

    fun setView(view: View): DialogUtils {
        mView = view
        dialogBuilder.setView(view)
        return this
    }

    fun setView(layoutRes: Int): DialogUtils {
        mView = (context as AppCompatActivity).layoutInflater.inflate(layoutRes, null)
        dialogBuilder.setView(mView)
        return this
    }

    fun setTitle(title: String): DialogUtils {
        dialog!!.setTitle(title)
        return this
    }

    fun setMessage(message: String): DialogUtils {
        dialog!!.setMessage(message)
        return this
    }

    fun show(): DialogUtils {
        dialog!!.show()
        if (positiveButton == null)
            positiveButton = dialog!!.getButton(DialogInterface.BUTTON_POSITIVE)
        if (negativeButton == null)
            negativeButton = dialog!!.getButton(DialogInterface.BUTTON_NEGATIVE)
        if (mNeutral == null)
            mNeutral = dialog!!.getButton(DialogInterface.BUTTON_NEUTRAL)
        return this
    }

    fun cancel(): DialogUtils {
        dialog!!.cancel()
        return this
    }

    fun showConfirmDialog(
        retainView: Boolean,
        icRes: Int,
        title: String?,
        message: String?,
        cancelText: String?,
        confirmText: String?,
        confirmListener: OnClickListener?
    ): DialogUtils {
        if (dialog == null) create()
        show()
        mNeutral!!.visibility = View.GONE
        if (mView != null && !retainView) mView!!.visibility = View.GONE
        if (title != null) dialog!!.setTitle(title) else dialog!!.setTitle("")
        if (message != null) dialog!!.setMessage(message) else dialog!!.setMessage("")
        if (cancelText != null) negativeButton!!.text =
            cancelText else negativeButton!!.visibility = View.GONE
        if (confirmText != null) positiveButton!!.text =
            confirmText else positiveButton!!.visibility = View.GONE
        if (icRes != 0) dialog!!.setIcon(icRes) else dialog!!.setIcon(null)
        if (confirmListener != null)
            positiveButton!!.setOnClickListener { confirmListener.onClick(dialog) }

        return this
    }

    interface OnClickListener {
        fun onClick(p1: AlertDialog?)
    }

    companion object {

        @SuppressLint("InflateParams")
        fun showEditTextDialog(context: Context, title: String?, message: String?, editorContent: String?, description: String?, confirmText: String?, confirmListener: OnClickListener): EditText {
            val contentView = (context as AppCompatActivity).layoutInflater.inflate(
                R.layout.edit_text,
                null
            ) as LinearLayout
            val edtTxt = contentView.findViewById(R.id.edt_txt) as EditText
            edtTxt.setText(editorContent)
            edtTxt.hint = description
            edtTxt.setSelectAllOnFocus(true)
            edtTxt.setSingleLine(true)
            val dialog = AlertDialog.Builder(context)
                .setTitle(title)
                .setMessage(message)
                .setView(contentView)
                .setPositiveButton(confirmText, null)
                .setNegativeButton("cancel", null)
                .create()
            dialog.window!!.attributes.windowAnimations = R.style.DialogTheme
            dialog.show()
            dialog.getButton(DialogInterface.BUTTON_POSITIVE)
                .setOnClickListener { confirmListener.onClick(dialog) }

            return edtTxt
        }

        @SuppressLint("InflateParams")
        fun showTermDialog(context: Context, title: String?, subtitle: String?, positive: String?,
                           negative: String?): Array<Any> {
            val titleView = (context as AppCompatActivity).layoutInflater.inflate(
                R.layout.term_dialog_header,
                null
            ) as LinearLayout
            (titleView.findViewById(R.id.term_title) as TextView).text = title
            (titleView.findViewById(R.id.term_description) as TextView).text = subtitle

            val contentView =
                context.layoutInflater.inflate(R.layout.term_dialog_content, null) as LinearLayout
            contentView.findViewById<ScrollView>(R.id.term_scroller).layoutParams = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                (MainActivity.SCREEN_HEIGHT * 0.5).toInt()
            )
            val dialog = AlertDialog.Builder(context)
                .setCustomTitle(titleView)
                .setView(contentView)
                .setNegativeButton(negative, null)
                .setPositiveButton(positive, null)
                .create()
            dialog.window!!.attributes.windowAnimations = R.style.DialogTheme
            dialog.show()
            dialog.setCancelable(false)
            dialog.getButton(DialogInterface.BUTTON_POSITIVE).visibility = View.GONE
            dialog.getButton(DialogInterface.BUTTON_NEGATIVE).isEnabled = false

            return arrayOf(
                dialog,
                contentView.findViewById(R.id.term_text),
                contentView.findViewById(R.id.term_progress)
            )
        }


        fun showConfirmDialog(context: Context,
            title: String?, content: String?,
            spannedContent: Spanned?, confirmTxt: String?,
            listener: OnClickListener?): AlertDialog {
            val dialogBuilder = AlertDialog.Builder(context).setTitle(title)
                .setMessage(content)
                .setNegativeButton("cancel", null)
                .setPositiveButton(confirmTxt, null)
            if (spannedContent != null) {
                val tw = TextView(context)
                tw.setPadding(40, 15, 20, 0)
                tw.text = spannedContent
                dialogBuilder.setView(tw)
            }
            val dialog = dialogBuilder.create()
            dialog.window!!.attributes.windowAnimations = R.style.DialogTheme
            dialog.show()
            if (listener != null) {
                dialog.getButton(DialogInterface.BUTTON_POSITIVE)
                    .setOnClickListener { listener.onClick(dialog) }
            }
            return dialog
        }
    }
}
